lib/bolt/node/winrm.rb

```diff
@@ -12,9 +12,17 @@ def initialize(host, port, user, password, shell: :powershell, **kwargs)
     end
 
     def connect
-      @connection = ::WinRM::Connection.new(endpoint: @endpoint,
-                                            user: @user,
-                                            password: @password)
+      options = { endpoint: @endpoint,
+                  user: @user,
+                  password: @password,
+                  retry_limit: 1 }
+
+      if @timeout
```

Comment:
Is this both connection and execution timeouts? Why are we changing from the default of `receive_timeout` being 10 seconds longer than `operation_timeout`?

Reply:
Also, why are we changing the `retry_limit`?

Reply:
I need to play a bit more with `receive_timeout` to see if it actually influences `operation_timeout` in a meaningful way. The initial defaults seemed a bit arbitrary based on observed behavior.

`retry_limit` is what causes the `timeout` to be completely ineffectual - i.e. what happens by default (with `retry_limit` of 3) is:

* fails to connect
* sleep for 10 seconds (default retry sleep)
* retry
* fails to connect
* sleep for 10 seconds
* retry
* fails to connect
* retry limit hit, fail

So for a connection failure, by default, it takes 20+ seconds before it fails, which doesn't make a lot of sense when the intent is to wait up to 1 second for a fail.

There are some notes in the commit about this, and I'll add more detail when this is finished.